src/index.js

```diff
@@ -1,25 +1,34 @@
-import { useState, useEffect } from "react";
+import React, { useState, useRef, useEffect } from "react";
 
 const noop = () => undefined;
 export default function useInputDebounce(
   effect = noop,
-  { initial, delay = 0, minLength = 0 }
+  { initial = "", delay = 0, minLength = 0 }
 ) {
   const [value, setValue] = useState(initial);
+  const debounceEnabled = useRef(true);
   const [onChange] = useState(() => e => {
-    const value = e && e.target ? e.target.value : e;
-    setValue(value || "");
+    const value = e && e.target ? e.target.value : undefined;
+    if (value) setValue(value);
```

Comment:
I left this in as I don't think we want to set the state "value" to 'undefined'?  Let me know if you would rather it just set to whatever it gets back from the line above?

Reply:
We do not need to have it anymore because we are using `onChange` only in the input element which will always has a e.target.value . Does it make sense?
``` 
const [onChange] = useState(() => e => {
    setValue(e.target.value);
  }); 
```

Reply:
Correct you are, didn't think of that now that we are using updateValue instead. 👍🏼